src/core/runtime/runtime.inl

```diff
@@ -35,8 +35,15 @@ void invoke_legate_registration_callback(Legion::Machine,
 template <Core::RegistrationCallback CALLBACK>
 /*static*/ void Core::perform_registration()
 {
-  Legion::Runtime::perform_registration_callback(
-    detail::invoke_legate_registration_callback<CALLBACK>, true /*global*/);
+  auto runtime = Runtime::get_runtime();
+  if (runtime->is_in_callback())
```

Comment:
What would cause multiple callbacks to get invoked recursively? Also - can we guarantee that `perform_registration_callback` will actually invoke the callback? Doesn't it sometimes defer the callback?

If Legion defers the callback until later, it might invoke the callback when `runtime->is_in_callback` is false?